soroban-env-host/src/events/mod.rs

```diff
@@ -26,9 +26,15 @@ pub struct HostEvent {
 fn display_address(addr: &ScAddress, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
     match addr {
         ScAddress::Account(acct) => match &acct.0 {
-            PublicKeyTypeEd25519(e) => write!(f, "Address(Account({}))", e),
+            PublicKeyTypeEd25519(e) => {
+                let strkey = stellar_strkey::ed25519::PublicKey(e.0.clone());
+                write!(f, "Address(Account({}))", strkey)
```

Comment:
What is this used for? Is the `Address(Account(` required? Asking because the `Address(Account(` seems redundant, that same information will be obvious in the strkey.

Reply:
At the very least we can drop the `Account(` part I think.

Reply:
Yeah, I suggest just display addresses as strkeys everywhere - there shouldn't be contexts where we actually want to communicate e.g. AccountId without the corresponding address.

Reply:
To be clear, the same goes for the contract address.

Reply:
Nothing is required, this is purely for display purposes; I put it there originally because they were uninformative 32-byte hex blobs. Maybe if strkeys are more-obvious to users we can remove it. I have no strong feelings.